### PR TITLE
Rework Poly example on pydpf

### PR DESCRIPTION
**Pull request by MichaelNale**
Rework poly example on pydpf using /fluent/3D/Polys.cas.h5 mesh

### CODE REVIEW

**Comment by rafacanton** [comment]
@MichaelNale Instead of adding a new file, can you directly rework the `pydpf-core/examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py`?

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 1]
```diff
@@ -1,108 +1,380 @@
-# noqa: D400
```
Comment (rafacanton) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:1]: Please keep formatting comments for the users

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 2]
```diff
@@ -1,108 +1,380 @@
...
-    [0.02, 0.01, 0.02],
-    [0.03, 0.01, 0.02],
-    [0.035, 0.005, 0.02],
-    [0.03, 0.0, 0.02],
-    [0.02, 0.0, 0.03],
-    [0.02, 0.01, 0.03],
-    [0.03, 0.01, 0.03],
-    [0.035, 0.005, 0.03],
-    [0.03, 0.0, 0.03],
+    [0, 0, 0],  # 0
```
Comment (rafacanton) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:19]: Please eliminate unneeded nodes

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 3]
```diff
@@ -1,108 +1,380 @@
...
 ]
 
-coordinates = polygon_points + polyhedron_points
-
 ###############################################################################
 # Create a bare mesh with pre-reserved memory
-mesh = dpf.MeshedRegion(num_nodes=len(coordinates), num_elements=2)
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+mesh_SolidOnly = dpf.MeshedRegion(num_nodes=len(polyhedron_points), num_elements=1)
```
Comment (rafacanton) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:53]: rework num_elements and num_nodes for each mesh

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 4]
```diff
@@ -1,108 +1,380 @@
...
-# Define the faces connectivity
-polyhedron_faces_connectivity = [
-    [5, 6, 7, 8, 9],
-    [5, 6, 11, 10],
-    [5, 9, 14, 10],
-    [9, 14, 13, 8],
-    [8, 13, 12, 7],
-    [7, 12, 11, 6],
-    [10, 11, 12, 13, 14],
+# Set the ELEMENT_FACE reverse scoping
```
Comment (rafacanton) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:234]: propertyfield, not scoping

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 5]
```diff
@@ -1,108 +1,380 @@
...
+for element_nodes_index_solid, element_nodes_solid in enumerate(polyhedron_element_node_connectivity):
+    connectivity_e_n_solid.append(element_nodes_solid, element_nodes_index_solid)
+mesh_SolidOnly.set_property_field(property_name="elements_nodes_connectivity", value=connectivity_e_n_solid)
+
+###############################################################################
+# Set the element/face type into a propertyfield
+
+"""ShellOnly"""
+# face types
+FT_Shell = [[dpf.element_types.Line2.value]]
```
Comment (rafacanton) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:336]: `FT_Shell = [dpf.element_types.Line2.value] * num`

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 6]
```diff
@@ -1,108 +1,474 @@
...
-polyhedron_faces_connectivity = [
-    [5, 6, 7, 8, 9],
-    [5, 6, 11, 10],
-    [5, 9, 14, 10],
-    [9, 14, 13, 8],
-    [8, 13, 12, 7],
-    [7, 12, 11, 6],
-    [10, 11, 12, 13, 14],
+"""
+Now also have to define a PropertyField relative the element_face reverse
```
Comment (rafacanton) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:225]: I don't think this sentence is clear enough. I would rephrase it to something like:

In addition to the relative previous `PropertyFields`, it is necessary to define the orientation of the faces relative to the cells that they surround. A given face has its nodal connectivity defined in a certain order, which means that the two cells that it bounds are either at the positive or negative sides of the face. We indicate this with the `elements_faces_reversed` PropertField, whose values can be either 0 or 1. By convention, 0 means that the face->node connectivity implies that the given cell is at side 0 of the face, and 1 the opposite.

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 7]
```diff
@@ -1,32 +1,31 @@
...
 Create and display a mesh with polygon and polyhedron elements
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 This example shows how to manually create a
 :class:`MeshedRegion <ansys.dpf.meshed_region.MeshedRegion>`
-object with two elements, a polygon and a polyhedron.
+object with many elements, polygons and a polyhedrons.
+
+.. note::
+    Because the cell_faces connectivity and the faces_connectivity give all the information concerning the mesh
+    we decide to avoid the process of adding element by defining directly the PropertyField.
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:12]: A bit unclear to me. Maybe rephrase? With short and simple sentences.

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 8]
```diff
@@ -1,32 +1,31 @@
...
-"""
-
-
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
-"""NODE COORDINATE"""
-"""Note on these two sets of nodes, there are two nodes at [0,0,0]
-This anomaly has been taking into account for this example"""
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Define manually the node coordinates of the polyhedrons and polygons
+# ~~~~~~~~~~~~~~~~~~~~
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:24]:
```suggestion
# Create new meshed regions from nodes
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 9]
```diff
@@ -1,32 +1,31 @@
...
+# Note on these two sets of nodes, there are two nodes at [0,0,0]
+# This anomaly has been taking into account for this example
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:26]: @MichaelNale I do not understand what you mean.

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 10]
```diff
@@ -1,32 +1,31 @@
...
 
+# Node coordinate of the polygons
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:28]:
```suggestion
# Node coordinates of the polygons
```

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 11]
```diff
@@ -49,6 +48,8 @@
     [-9.39159748e-36, -1.57209197e-20, 1.73472348e-18],     # 17
 ]
 
+# Node coordinate of the polyhedrons
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:51]:
```suggestion
# Node coordinates of the polyhedrons
```

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 12]
```diff
@@ -72,52 +73,40 @@
 ]
 
 ###############################################################################
-"""
-We use these coordinates to build meshed region
-We define here a meshed region relative to the shell (nodes -> lines -> surfaces) and one to the solid (nodes -> surfaces -> volumes)
-"""
-
+# We use these coordinates to build meshed region
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:76]:
```suggestion
# We then use these coordinates to build two meshed regions.
```

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 13]
```diff
@@ -72,52 +73,40 @@
...
+# We define here a meshed region relative to the shell (nodes -> lines -> surfaces)
+# and one to the solid (nodes -> surfaces -> volumes)
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:78]:
```suggestion
# One meshed region will define shell elements and their connectivity (nodes -> lines -> surfaces),
# while the second will define solid elements and their own connectivity (nodes -> surfaces -> volumes).
```

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 14]
```diff
@@ -72,52 +73,40 @@
...
 
 
 ###############################################################################
-"""We define now all the connectivity for the two meshed regions"""
-
-
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
-"""FACE_NODE CONNECTIVITY"""
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# We define now all the connectivity for the two meshed regions
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:104]:
```suggestion
# Create the connectivity tables
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 15]
```diff
@@ -72,52 +73,40 @@
...
 
+###############################################################################
+# face node connectivity
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:107]:
```suggestion
# face-to-node connectivity
```

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 16]
```diff
@@ -184,11 +173,10 @@
 ]
 
 
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
-"""CELL_FACE CONNECTIVITY"""
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+###############################################################################
+# cell_face connectivity
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:177]:
```suggestion
# cell-to-face connectivity
```

**Review by rafacanton — APPROVED** [review]

File: examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py [hunk 17]
```diff
@@ -1,108 +1,448 @@
...
+# Now also have to define a PropertyField relative the element_face reverse.
+# It is necessary to define the orientation of the faces relative to the cells
+# that they surround.
+# A given face has its nodal connectivity defined in a certain order,
+# which means that the two cells that it bounds are
+# either at the positive or negative sides of the face. We indicate this with
+# the elements_faces_reversed PropertyField,
+# # whose values can be either 0 or 1.
+# By convention, 0 means that the face->node connectivity implies that the given
+# cell is at side 0 of the face, and 1 the opposite
```
Comment (PProfizi) [inline examples/09-mesh_operations/00-create_polygons_and_polyhedrons.py:222]: @MichaelNale no mention of normals or how this orientation is computed?
